src/app/tests/suites/TestModeSelectCluster.yaml

```diff
@@ -88,3 +91,23 @@ tests:
                 value: 2
       response:
           error: CONSTRAINT_ERROR
+
+    - label: "Change to Unsupported StartUp Mode"
+      command: "writeAttribute"
+      attribute: "StartUpMode"
+      arguments:
+        value: 2
+      response:
+        error: CONSTRAINT_ERROR
+
+    - label: "Change to Supported StartUp Mode"
```

Comment:
This is not testing the actual startup behavior.  Please add tests for that?

Reply:
This would require using the System's Reboot command, which was pointed out does not run locally with chip-tool in this [slack conversation](https://csamembers.slack.com/archives/G014G30SVV0/p1646854997668579). 
If this is for CI testing, I can include it in another PR, but it raises the question of how should I test it locally?

Reply:
You test it locally by doing:
```
./scripts/run_in_build_env.sh "./scripts/tests/run_test_suite.py --chip-tool /path/to/chip-tool --target TestModeSelectCluster run --iterations 1 --all-clusters-app /path/to/all-clusters-app"
```
which is what CI runs.

Reply:
This is great. I've included the test in the new push